client/extensions/woocommerce/app/settings/taxes/index.js

```diff
@@ -9,207 +9,92 @@ import PropTypes from 'prop-types';
 import { bindActionCreators } from 'redux';
 import classNames from 'classnames';
 import { connect } from 'react-redux';
+import { find, isEmpty } from 'lodash';
 import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-import ActionHeader from 'woocommerce/components/action-header';
-import {
-	areSettingsGeneralLoaded,
-	areTaxCalculationsEnabled,
-} from 'woocommerce/state/sites/settings/general/selectors';
-import {
-	areTaxSettingsLoaded,
-	getPricesIncludeTax,
-	getShippingIsTaxFree,
-} from 'woocommerce/state/sites/settings/tax/selectors';
-import ExtendedHeader from 'woocommerce/components/extended-header';
-import { updateTaxesEnabledSetting } from 'woocommerce/state/sites/settings/general/actions';
-import QuerySettingsGeneral from 'woocommerce/components/query-settings-general';
-import { fetchTaxRates } from 'woocommerce/state/sites/meta/taxrates/actions';
-import { fetchTaxSettings, updateTaxSettings } from 'woocommerce/state/sites/settings/tax/actions';
-import { getLink } from 'woocommerce/lib/nav-utils';
+import { fetchPlugins } from 'state/plugins/installed/actions';
+import { getPlugins, isRequesting } from 'state/plugins/installed/selectors';
 import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
 import Main from 'components/main';
-import TaxSettingsSaveButton from './save-button';
-import SettingsNavigation from '../navigation';
-import { successNotice, errorNotice } from 'state/notices/actions';
-import StoreAddress from 'woocommerce/components/store-address';
-import TaxesOptions from './taxes-options';
-import TaxesRates from './taxes-rates';
+import SettingsTaxesPlaceholder from './taxes-placeholder';
+import SettingsTaxesTaxJar from './taxes-taxjar';
+import SettingsTaxesWooCommerceServices from './taxes-wcs';
 
 class SettingsTaxes extends Component {
-	constructor( props ) {
-		super( props );
-		this.state = {
-			isSaving: false,
-			pricesIncludeTaxes: props.pricesIncludeTaxes,
-			shippingIsTaxable: props.shippingIsTaxable,
-			taxesEnabled: props.taxesEnabled,
-			userBeganEditing: false,
-		};
-	}
-
 	static propTypes = {
-		site: PropTypes.shape( {
-			slug: PropTypes.string,
-			ID: PropTypes.number,
-		} ),
 		className: PropTypes.string,
+		isRequestingSitePlugins: PropTypes.bool,
+		siteId: PropTypes.number,
+		sitePluginsLoaded: PropTypes.bool,
+		siteSlug: PropTypes.string,
+		taxJarPluginActive: PropTypes.bool,
 	};
 
 	componentDidMount = () => {
-		const { site } = this.props;
+		const { isRequestingSitePlugins, siteId, sitePluginsLoaded } = this.props;
```

Comment:
This logic is duplicated in `componentWillReceiveProps`. Perhaps it can be extracted into a `maybeFetchPlugins` method and called in both lifecycle methods?

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/19535/commits/4dc13074b82efbf815f60cf9475d4ecdb8eaf096